### DISCUSSION TITLE
Color vs SolidColorBrush for a StyledProperty

### DISCUSSION THREAD

**Discussion by LaurentInSeattle** [AvaloniaUI/Avalonia#15581] [Q&A] [answered]
Hello everyone, 

I am very new to Avalonia and I am trying to port a user control originally designed for WPF 
The WPF control was taking a SolidColorBrush for one of its dependency property, named "Foreground".
When I do something similar with Avalonia with a StyledProperty, the control receives an Immutable brush object (and fails) when setting 
something like Foreground="DarkOrange" when using the control in a view.  

I am now using Color instead of brushes and the control does work but setting up the color is somewhat annoying for client code: 
`        <ctrl:SvgIcon
            Source="power_button"
            HorizontalAlignment="Right" VerticalAlignment="Top"
            Margin="12 64 12 12"
            Wadding="4"
            Height="32" Width="32"
            StrokeThickness="1"
            >
            <ctrl:SvgIcon.Foreground>
                <Color>DarkOrchid</Color>
            </ctrl:SvgIcon.Foreground>
        </ctrl:SvgIcon>
`
  
Any suggestions ? Or am I doing something wrong with the brushes ?
Thanks ! 

Complete solution for .net 8 is here: https://github.com/LaurentInSeattle/AvaloniaExperiments 
SvgIcon is the troublemaker...

**Comment by stevemonaco** [comment, marked as answer]
The `Background` and `Foreground` dependency properties should take `IBrush` instead of `Brush` or `Color`. Then just use it like normal, eg:
```xml
<ctrl:SvgIcon Foreground="#FF0000" />
<ctrl:SvgIcon Foreground="Red" />
```

**Reply by LaurentInSeattle** [reply]
Thanks, that's work...

**Reply by timunie** [reply]
> [!TIP] 
> We also have PathIcon which can take the SVGs path data if that helps.

**Reply by LaurentInSeattle** [reply]
"My" SvgIcon goes a little step further and can work with DrawingImage, which is the format provided by the SharpVectors library when converting svg files. The generated xaml is almost the same between WPF and Avalonia.
